src/opensearch.ts

```diff
@@ -1,159 +1,278 @@
-import { pipe } from 'fp-ts/lib/pipeable'
-import { chain, fromOption, TaskEither, tryCatch } from 'fp-ts/lib/TaskEither'
-import { ExtractedResult, extractInnerHits, extractOuterHits } from './extract'
-import { ApplicationError, StatusCodes } from './http'
-import { Balance, BalanceValue, Block, Hash, OpenSearchBlock, OpenSearchSnapshot, Ordinal, RewardTransaction, Snapshot, WithRewards, WithTimestamp } from './model'
-import { fieldTerm, getByFieldNestedQuery, getLatestQuery, getQuery } from './queries'
-
-const OSIndex = {
-    Snapshots: process.env.OPENSEARCH_INDEX!
-}
-
-export const getClient = (): Client => {
-    return new Client({ node: process.env.OPENSEARCH_NODE })
-}
-
-export const getSnapshot = (os: Client) => (predicateName: string, predicateValue: string = 'latest'): TaskEither<ApplicationError, Snapshot> => {
-
-    const outerIncludes = ['ordinal', 'hash', 'height', 'subHeight', 'timestamp', 'blocks']
-    const osSearch = getSnapshotQuery<OpenSearchSnapshot>(predicateName, predicateValue, outerIncludes)(os)
+import { ApiResponse, Client } from '@opensearch-project/opensearch';
+import {
+  QueryDslNestedQuery,
+  SearchResponse,
+} from '@opensearch-project/opensearch/api/types';
+import { TransportRequestPromise } from '@opensearch-project/opensearch/lib/Transport';
+import * as O from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { chain, fromOption, TaskEither, tryCatch } from 'fp-ts/lib/TaskEither';
+import { ExtractedResult, extractInnerHits, extractOuterHits } from './extract';
+import { ApplicationError, StatusCodes } from './http';
+import {
+  Balance,
+  BalanceValue,
+  Block,
+  Hash,
+  OpenSearchBlock,
+  OpenSearchSnapshot,
+  Ordinal,
+  RewardTransaction,
+  Snapshot,
+  WithRewards,
+  WithTimestamp,
+} from './model';
+import {
+  fieldTerm,
+  getByFieldNestedQuery,
+  getLatestQuery,
+  getQuery,
+} from './queries';
 
-    return pipe(findOuter(osSearch), chain(extractSnapshot))
+enum OSIndex {
+  Snapshots = 'snapshots',
+  Blocks = 'blocks',
+  Transactions = 'transactions',
+  Balances = 'balances',
 }
 
-export const getSnapshotRewards = (os: Client) => (predicateName: string, predicateValue: string = 'latest'): TaskEither<ApplicationError, RewardTransaction[]> => {
+export const getClient = (): Client => {
+  return new Client({ node: process.env.OPENSEARCH_NODE });
+};
 
-    const outerIncludes = ['rewards']
-    const osSearch = getSnapshotQuery<WithRewards>(predicateName, predicateValue, outerIncludes)(os)
+export const getSnapshot =
+  (os: Client) =>
+  (
+    predicateName: string,
+    predicateValue: string | number | 'latest' = 'latest'
+  ): TaskEither<ApplicationError, Snapshot> => {
+    const outerIncludes: (keyof Snapshot)[] = [
+      'ordinal',
+      'hash',
+      'height',
+      'subHeight',
+      'timestamp',
+      'blocks',
+    ];
+    const osSearch = getSnapshotQuery<OpenSearchSnapshot>(
+      predicateName,
+      predicateValue,
+      []
```

Comment:
Shouldn't be here `outerIncludes` not empty array?

Reply:
you are right, I missed that.